src/main/java/de/tum/in/www1/artemis/service/LearningPathService.java

```diff
@@ -178,17 +181,46 @@ private void updateLearningPathProgress(@NotNull LearningPath learningPath) {
      */
     public LearningPathHealthDTO getHealthStatusForCourse(@NotNull Course course) {
         if (!course.getLearningPathsEnabled()) {
-            return new LearningPathHealthDTO(LearningPathHealthDTO.HealthStatus.DISABLED);
+            return new LearningPathHealthDTO(Set.of(LearningPathHealthDTO.HealthStatus.DISABLED));
+        }
+
+        Set<LearningPathHealthDTO.HealthStatus> status = new HashSet<>();
+        Long numberOfMissingLearningPaths = checkMissingLearningPaths(course, status);
+        checkNoCompetencies(course, status);
+        checkNoRelations(course, status);
+
+        // if no issues where found, add OK status
+        if (status.isEmpty()) {
+            status.add(LearningPathHealthDTO.HealthStatus.OK);
         }
 
+        return new LearningPathHealthDTO(status, numberOfMissingLearningPaths);
+    }
+
+    private Long checkMissingLearningPaths(@NotNull Course course, @NotNull Set<LearningPathHealthDTO.HealthStatus> status) {
         long numberOfStudents = userRepository.countUserInGroup(course.getStudentGroupName());
         long numberOfLearningPaths = learningPathRepository.countLearningPathsOfEnrolledStudentsInCourse(course.getId());
+        Long numberOfMissingLearningPaths = numberOfStudents - numberOfLearningPaths;
 
-        if (numberOfStudents == numberOfLearningPaths) {
-            return new LearningPathHealthDTO(LearningPathHealthDTO.HealthStatus.OK);
+        if (numberOfMissingLearningPaths != 0) {
+            status.add(LearningPathHealthDTO.HealthStatus.MISSING);
         }
         else {
-            return new LearningPathHealthDTO(LearningPathHealthDTO.HealthStatus.MISSING, numberOfStudents - numberOfLearningPaths);
+            numberOfMissingLearningPaths = null;
```

Comment:
why do you not just set it to 0 and are able to use a primitive type?

Reply:
see comment above